DownFall/Extensions/CGPoint+Extension.swift

```diff
@@ -111,25 +126,60 @@ extension CGPoint {
         let y: CGFloat
         switch verticaliy {
         case .top:
-            y = that.center.y + that.height/2 + this.height/2 + yOffset
+            y = that.maxY + this.height/2 + yOffset
         case .center:
             y = that.center.y
         case .bottom:
-            y = that.center.y - that.height - this.height/2 - yOffset
+            y = that.minY - this.height/2 - yOffset
         }
         
         let x: CGFloat
         switch anchor {
         case .left:
-            x = that.center.x - that.width/2 - this.width/2 - xOffset
+            x = that.minX - this.width/2 - xOffset
         case .right:
-            x = that.center.x + that.width/2 + this.width/2 + xOffset
+            x = that.maxX + this.width/2 + xOffset
+        case .center:
+            x = that.center.x
```

Comment:
We should go thru all the method in this extension and see if we can condense them into the align methods and the position inside methods.  It seems like those cover the majority of use cases and we can provide convenience funds on top of those